### PR TITLE
Dependencies not in visible section causing OutOfBounds exception

### PR DESCRIPTION
**Pull request by jacobvanorder**
According to Bug #3222, the commuter section wasn't displaying sections correctly. In delving into the issue more, it seemed to be a two-pronged issue where NUResponseSet wasn't creating proper dependency graphs due to writing over the container array for which items to hide and show. The second issue was related to inserting in the dependent questions that were to appear into the section's tableview. Often, the index would be wrong or, worse, out of bounds of the data source holding the visible sections. This was resolved, after much trial and error, by checking against NSNotFound. 

I compiled a test for the example app called "dependency toolbox" that mimics the main issues I ran into: dependencies in another section, dependency with multiple answers and multiple dependencies from an answer.

### CODE REVIEW

File: Frank/features/basic_survey.feature [hunk 1]
```diff
@@ -5,13 +5,13 @@ Feature:
 
   Scenario:
     Basic questions
-  Given I launch the app using iOS 5.1 and the ipad simulator
+  Given I launch the app using iOS 6.1 and the ipad simulator
```
Comment (jdzak) [inline Frank/features/basic_survey.feature]: I would suggest leaving the targeted simulator at 5.1 for now, unless there is a reason to change it.

File: Frank/features/my_first.feature [hunk 2]
```diff
@@ -1,11 +1,11 @@
-Feature:
+Feature:
```
Comment (jdzak) [inline Frank/features/my_first.feature]: Watch out for rogue white-space in commits.

File: Frank/frank_static_resources.bundle/index.html [hunk 3]
```diff
@@ -12,10 +12,8 @@
```
Comment (jdzak) [inline Frank/frank_static_resources.bundle/index.html:11]: I'm guessing the Frank assets where updated as a result of running `frank setup`?

File: Frank/features/basic_survey.feature (hunk 1 above)
Comment (jacobvanorder) [inline Frank/features/basic_survey.feature]: Yeah, sorry. I didn't download the 5.1 simulator and did that in order to get up and running quickly.

File: Frank/frank_static_resources.bundle/index.html (hunk 3 above)
Comment (jacobvanorder) [inline Frank/frank_static_resources.bundle/index.html:11]: Maybe? It wasn't intentional…

File: KIF/SectionTVCTest.h [hunk 4]
```diff
@@ -6,8 +6,8 @@
 //  Copyright (c) 2012 Northwestern University. All rights reserved.
 //
 
-#import "KIFTestController.h"
+//#import "KIFTestController.h"
 
-@interface SectionTVCTest : KIFTestController
+@interface SectionTVCTest /*: KIFTestController*/
```
Comment (jdzak) [inline KIF/SectionTVCTest.h]: This looks like it was checked in by accident.

Comment (jacobvanorder) [inline KIF/SectionTVCTest.h]: Yeah. The app won't build without that as there is no KIFTestController. Was this something that was attempted and reversed?

File: Frank/frankified_build/NUSurveyorExample.app/PkgInfo [hunk 5]
```diff
@@ -0,0 +1 @@
```
Comment (jdzak) [inline Frank/frankified_build/NUSurveyorExample.app/PkgInfo]: The Frank/frankified_build looks like it's artifacts generated from building the NUSurveyorExampleApp via frank. We shouldn't need to check any of that code in, so I think we can safely delete the Frank/frankified_build directory.

File: NUSurveyor/Controllers/NUSectionTVC.m [hunk 6]
```diff
@@ -719,13 +723,14 @@ - (void) showAndHideDependenciesTriggeredBy:(NSIndexPath *)idx {
 	//  DLog(@"showHide: %@", showHide);
   for (NSString *question in [showHide objectForKey:@"show"]) {
     // show the question and insert it in the right place
-    if (![self findVisibleSectionWithUUID:question]) {
+    if ([self findVisibleSectionWithUUID:question] == nil) {
       NSUInteger i = [self indexForInsert:question];
       if (i != 0U) { // NSUInteger 0 is returned by indexForInsert if nothing is found
-        [[self.allSections objectAtIndex:[self indexOfQuestionOrGroupWithUUID:question]] setObject:NS_YES forKey:@"show"];
+          [[self.allSections objectAtIndex:[self indexOfQuestionOrGroupWithUUID:question]] setObject:NS_YES forKey:@"show"];
```
Comment (jdzak) [inline NUSurveyor/Controllers/NUSectionTVC.m]: Whitespace.

File: NUSurveyor/NSString+NUAdditions.m [hunk 7]
```diff
@@ -5,6 +5,7 @@ @implementation NSString (NUAdditions)
 static NSString* SPACE = @" ";
 
 - (NSString*) normalizeWhitespace {
+    [SPACE description];
```
Comment (jdzak) [inline NUSurveyor/NSString+NUAdditions.m]: Accidentally committed?

Comment (jacobvanorder) [inline NUSurveyor/NSString+NUAdditions.m]: This was to quiet a debugger warning as it viewed the SPACE as a unused variable which is is within the scope of the class. I just did a description to quiet the debugger.

Comment (jdzak) [inline NUSurveyor/NSString+NUAdditions.m]: Oh, if it's not used we can remove it.

**Comment by jdzak** [comment]
One more thing, can we squash all these commits into one? While on the commuterLogicFix branch `rebase -i master` should allow you to squash all the commits into one.

**Comment by jacobvanorder** [comment]
Done!

**Pull request merged**